Debug App/Tests/Unit Tests/Analytics/AnalyticsTests.swift

```diff
@@ -1025,241 +582,35 @@ class AnalyticsTests: XCTestCase {
         }
         
         wait(for: expectationsToBeFulfilled, timeout: 60)
-        storedEvents = (try? Analytics.Service.loadEventsSynchronously()) ?? []
+        storedEvents = Analytics.Service.loadEvents()
         let nonNetworkEvents = storedEvents.filter({ $0.eventType != .networkCall && $0.eventType != .networkConnectivity })
         XCTAssert(nonNetworkEvents.count == 0, "nonNetworkEvents: \(nonNetworkEvents.count)")
     }
+}
+
+
+extension Promise {
+    static func fulfilled(_ value: T) -> Promise<T> {
+        return Promise<T> { $0.fulfill(value) }
+    }
     
-    func test_race_conditions_on_creating_events_and_deleting_analytics_file() throws {
```

Comment:
This test and the one below tested concurrency by creating various types of queue and calling the `save` and `delete` functions in different combination

I've removed concurrency concerns from those methods so they now do what they say on the tin. As such, we no longer need these tests and I'll instead add more tests to exercise concurrency when calling `record` / `flush` / `sync`